hilti/toolchain/src/compiler/jit.cc

```diff
@@ -605,6 +632,42 @@ Result<Nothing> JIT::JobRunner::_waitForJobs() {
     return Nothing();
 }
 
+void JIT::JobRunner::_recordUserDiagnostics(JobID jid, const Job& job) {
+    auto add_header = ! rt::filesystem::exists(UserDiagnosticsFile);
+
+    if ( add_header ) {
+        // Record version of C++ compiler.
+        reproc::options options;
+        options.redirect.path = UserDiagnosticsFile.c_str();
+        std::array<std::string, 2> cmdline = {hilti::configuration().cxx.c_str(), "--version"};
+        reproc::run(cmdline, options);
+    }
+
+    std::ofstream out(UserDiagnosticsFile, std::ios::app);
+    if ( ! out ) {
+        logger().warning(
+            util::fmt("could not open diagnostics file %s for writing: %s", UserDiagnosticsFile, ::strerror(errno)));
+        return;
+    }
+
+    // Copy the output of the process over into the diagnostics file
+    std::ifstream in(job.output);
+    if ( ! in ) {
+        logger().warning(
+            util::fmt("could not open process output file %s for reading: %s", job.output, ::strerror(errno)));
+        return;
+    }
+
+    if ( add_header )
+        out << "HILTI version " << hilti::configuration().version_string_long << "\n\n";
```

Comment:
Super-nit: Since this is using `add_header` again, I'd prefer it to be in the same block as the other `add_header` above, just to keep things together

Reply:
The thing is that we need to open `out` in between, in append mode, that's why it's split.